### PR TITLE
[Mono] Profile AOT: Relax is_local_inst to let more methods qualified for AOT compilation

### PR DESCRIPTION
**Pull request by fanyang-mono**
This PR allows more methods to be qualified for AOT compilation when using profiled AOT mode.

Specifically, relax `is_local_inst` to allow methods inflated with types from `System.Private.CoreLib.dll`.

Contributes to #101135

### The number of methods being AOT compiled before vs after this PR
**Test app**: the customer provided app https://github.com/crui3er/MauiAotTest with their own profile

**Test routine:**
1. Launch the app
2. Go to Counter and click 5 times
3. Go to Weather and wait for things to be loaded
4. Go back to Counter and click 5 times
5. Go back to Home page

In the device log:
**Before** this PR: 
- 973 hits of methods not being AOT'ed
- 605 hits of non-wrapper methods not being AOT'ed
- 6685 hits of methods got AOT'ed.

**After** this PR: 
- 817 hits of methods not being AOT'ed
- 448 hits of non-wrapper methods not being AOT'ed (Compared with before, ~26% more non-wrapper methods got AOT'ed)
- 6844 hits of methods got AOT'ed.

### App size and app startup time
**Test device:** Pixel 5
**Test app:** the customer provided app https://github.com/crui3er/MauiAotTest with their own profile
App startup time:
- Before: 641 ms
- After: 634 ms

App size:
- Before: 16MB
- After: 16MB

**Test app:** MAUI template app with default profile
App startup time:
- Before: 577.1 ms
- After: 573.4 ms

App size:
- Before: 15MB
- After: 15MB

In summary, there isn't significant change in terms of the app size and app startup time, as they are quite small apps.

### CODE REVIEW

**Comment by steveisok** [comment]
`DNCENGWIN-080` is having issues finding the device that should be attached to it. The android failure is an infra one.

**Review by lambdageek — APPROVED** [review]
